lib/backburner/queue.ts

```diff
@@ -155,30 +138,23 @@ export default class Queue {
     return false;
   }
 
-  private guidForTarget(target) {
-    if (!target) { return; }
-
-    let peekGuid = this.globalOptions.peekGuid;
-    if (peekGuid) {
-      return peekGuid(target);
-    }
-
-    let KEY = this.globalOptions.GUID_KEY;
-    if (KEY) {
-      return target[KEY];
-    }
-  }
-
-  private pushUniqueWithoutGuid(target, method, args, stack) {
-    let queue = this._queue;
+  public pushUnique(target, method, args, stack) {
+    let localQueue = this.targetQueues.get(target);
 
-    let index = findItem(target, method, queue);
-    if (index > -1) {
-      queue[index + 2] = args;  // replace args
-      queue[index + 3] = stack; // replace stack
+    if (localQueue === undefined) {
+      this.targetQueues.set(target, [
+        method,
+        this._queue.push(target, method, args, stack) - 4
```

Comment:
Can you remove this to a separate local variable? I know both of the original implementations did it this way, but (to me at least) having a separate variable named `indexIntoQueue` or something would make this a bit easier to grok.